app/javascript/react/components/ThresholdConfigurator/ThresholdButtons/UniformDistributionButton.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { useAppDispatch, useAppSelector } from "../../../store/hooks";
+import { RootState } from "../../../store";
+import { UniformDistributionIcon } from "./Icons/UniformDistributionIcon";
+import { ThresholdButton, ThresholdButtonVariant } from "./ThresholdButton";
+import { StatusEnum } from "../../../types/api";
+import { calculateUniformThresholds } from "../../../utils/uniformDistributionThresholdCalc";
+import { setUserThresholdValues } from "../../../store/thresholdSlice";
+import { useMapParams } from "../../../utils/mapParamsHandler";
+import { useTranslation } from "react-i18next";
+
+interface UniformDistributionButtonProps {
+  variant?: ThresholdButtonVariant;
+  swapIconTextPosition?: boolean;
+  useDarkBlueIcon?: boolean;
+  hasErrorMessage: (message: string) => void;
+}
+
+const UniformDistributionButton: React.FC<UniformDistributionButtonProps> = ({
+  variant = ThresholdButtonVariant.IconOnly,
+  swapIconTextPosition = false,
+  useDarkBlueIcon = false,
+  hasErrorMessage,
+}) => {
+  const dispatch = useAppDispatch();
+  const { sessionId } = useMapParams();
+  const { t } = useTranslation();
+
+  const mobileStream = useAppSelector((state: RootState) => state.mobileStream);
+  const fixedStream = useAppSelector((state: RootState) => state.fixedStream);
+
+  const onClick = () => {
+    let min = 0;
+    let max = 0;
+
+    if (
+      mobileStream.status === StatusEnum.Fulfilled &&
+      mobileStream.data.id === sessionId
+    ) {
+      min = Math.round(mobileStream.minMeasurementValue!);
+      max = Math.round(mobileStream.maxMeasurementValue!);
```

Comment:
I think we need to round the max value always to the higher one e.g. `Math.ceil`